### PR TITLE
Re-enable digest no-domain tests

### LINKED ISSUE #1045: DigestAuthentication_Echo_RoundTrips_String_No_Domain fails in the new test infrastructure
When run in the new test infrastructure, this test fails with the exception below when tested against an IIS hosted service.  Investigate and fix.  Probable cause is missing activation in IISHostedWcfService or mismatch of endpoint name
```
                        Http_ClientCredentialTypeTests.DigestAuthentication_Echo_RoundTrips_String_No_Domain [FAIL] (TaskId:1722)
                           System.ServiceModel.EndpointNotFoundException : There was no endpoint listening at http://wcfcoresrv4.cloudapp.net/wcftestservice1//HttpDigestNoDomain.svc// that could accept the message. This is often caused by an incorrect address or SOAP action. See InnerException, if present, for more details. (TaskId:1722)
                           Stack Trace: (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\HttpResponseMessageHelper.cs(325,0): at System.ServiceModel.Channels.HttpResponseMessageHelper.ValidateResponseStatusCode() (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\HttpResponseMessageHelper.cs(38,0): at System.ServiceModel.Channels.HttpResponseMessageHelper.<ParseIncomingResponse>d__7.MoveNext() (TaskId:1722)
                              --- End of stack trace from previous location where exception was thrown --- (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\HttpChannelFactory.cs(1018,0): at System.ServiceModel.Channels.HttpChannelFactory`1.HttpClientRequestChannel.HttpClientChannelAsyncRequest.<ReceiveReplyAsync>d__17.MoveNext() (TaskId:1722)
                              --- End of stack trace from previous location where exception was thrown --- (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\RequestChannel.cs(297,0): at System.ServiceModel.Channels.RequestChannel.<RequestAsync>d__34.MoveNext() (TaskId:1722)
                              --- End of stack trace from previous location where exception was thrown --- (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\RequestChannel.cs(258,0): at System.ServiceModel.Channels.RequestChannel.<RequestAsyncInternal>d__33.MoveNext() (TaskId:1722)
                              --- End of stack trace from previous location where exception was thrown --- (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) (TaskId:1722)
                                 at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\Internals\System\Runtime\TaskHelpers.cs(189,0): at System.Runtime.TaskHelpers.WaitForCompletion[TResult](Task`1 task) (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\RequestChannel.cs(247,0): at System.ServiceModel.Channels.RequestChannel.Request(Message message, TimeSpan timeout) (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Dispatcher\RequestChannelBinder.cs(105,0): at System.ServiceModel.Dispatcher.RequestChannelBinder.Request(Message message, TimeSpan timeout) (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannel.cs(764,0): at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout) (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannel.cs(721,0): at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs) (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannelProxy.cs(371,0): at System.ServiceModel.Channels.ServiceChannelProxy.InvokeService(MethodCall methodCall, ProxyOperationRuntime operation) (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\src\System\ServiceModel\Channels\ServiceChannelProxy.cs(139,0): at System.ServiceModel.Channels.ServiceChannelProxy.Invoke(MethodInfo targetMethod, Object[] args) (TaskId:1722)
                              --- End of stack trace from previous location where exception was thrown --- (TaskId:1722)
                                 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() (TaskId:1722)
                                 at System.Reflection.DispatchProxyGenerator.Invoke(Object[] args) (TaskId:1722)
                                 at generatedProxy_2.Echo(String ) (TaskId:1722)
                              c:\OSS\wcf\src\System.Private.ServiceModel\tests\Scenarios\Security\TransportSecurity\Http\ClientCredentialTypeTests.cs(32,0): at Http_ClientCredentialTypeTests.DigestAuthentication_Echo_RoundTrips_String_No_Domain() (TaskId:1722)
    
```

### PR DESCRIPTION
**Pull request by mconnew**
Fixes #1045

### CODE REVIEW

File: src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/AuthenticationResourceHelper.cs [hunk 1]
```diff
@@ -6,57 +6,56 @@
...
         {
-            public ResourceDigestServiceAuthorizationManager(string realm) : base(realm) { }
-            public override bool GetPassword(string username, out string password)
+            private const string DigestUsernameHeaderName = "DigestUsername";
+            private const string DigestPasswordHeaderName = "DigestPassword";
+            private const string DigestRealmHeaderName = "DigestRealm";
+
+            public ResourceDigestServiceAuthorizationManager() : base(string.Empty) { }
+
+            public override bool GetPassword(ref Message message, string username, out string password)
```
Comment (roncain) [inline src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/AuthenticationResourceHelper.cs]: Nit: this looks like it should probably be TryGetPassword.  Not blocking to leave as is.

**Comment by roncain** [comment]
@dotnet-bot test all outerloop please

**Comment by roncain** [comment]
This test fails on all the non-Windows OuterLoops, so I believe it needs [ConditionalFact(nameof(Digest_Authentication_Available)] to make it run only when CI or lab runs have explicitly enabled digest testing.  Look at how the other tests used #if FULLXUNIT_NOTSUPPORTED to give it conditional execution in NET Native.  Because of this I also recommend manually running "N on K" (run ToF against NET Native for this test).

**Comment by mconnew** [comment]
The issue on non-windows is because of an incorrect implementation of the digest authentication protocol. The point of this test is that we don't need any environmental things set up to verify digest authentication so we don't need a ConditionalFact.

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop please

**Comment by roncain** [comment]
LGTM

**Comment by iamjasonp** [comment]
LGTM

**Pull request merged**